### PR TITLE
ResourceAllocation - Initial DB & GET/POST/DELETE Endpoints

### PR DESCRIPTION
**Pull request by xssic**
- [x] New feature
- [ ] Bug fix
- [ ] High impact

**Description of work:**
Basic GET, PUT, POST, DELETE endpoints for ResourceAllocation

**Testing:**
- [ ] ~~Can be tested~~
- [x] Automatic tests created / updated
- [x] Local tests are passing

**Checklist:**
- [x] Considered automated tests
- [x] Considered updating specification / documentation
- [x] Considered work items 
- [x] Considered security
- [x] Performed developer testing
- [x] Checklist finalized / ready for review

### CODE REVIEW

File: src/backend/api/Fusion.Resources.Api/Startup.cs [hunk 1]
```diff
@@ -66,6 +66,7 @@ public void ConfigureServices(IServiceCollection services)
             services.AddOrgApiClient(Fusion.Integration.Org.OrgConstants.HttpClients.Application, Fusion.Integration.Org.OrgConstants.HttpClients.Delegate);
 
             services.AddControllers()
+                .AddNewtonsoftJson()
```
Comment (xssic) [inline src/backend/api/Fusion.Resources.Api/Startup.cs:69]: Required when adding ApiPropertiesCollection.

**Comment by HansDahle** [comment]
When doing multiple migrations in one PR, could we maybe merge them?

**Comment by xssic** [comment]
> When doing multiple migrations in one PR, could we maybe merge them?

Merged into one.

File: src/backend/api/Fusion.Resources.Database/Entities/DbResourceAllocationRequest.cs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+                entity.OwnsOne(e => e.ProvisioningStatus, op =>
+                {
+                    op.Property(ps => ps.State).HasConversion(new EnumToStringConverter<DbProvisionState>());
+                });
+                entity.OwnsOne(e => e.OrgPositionInstance);
+
+                entity.Property(e => e.Type).HasConversion(new EnumToStringConverter<DbAllocationRequestType>());
+                entity.Property(e => e.State).HasConversion(new EnumToStringConverter<DbRequestState>());
+                entity.Property(e => e.LastActivity).HasDefaultValue(DateTimeOffset.MinValue);
```
Comment (leifbjarte) [inline src/backend/api/Fusion.Resources.Database/Entities/DbResourceAllocationRequest.cs:57]: Can default be DateTimeOffset.UtcNow?

File: src/backend/api/Fusion.Resources.Api/Controllers/ResourceAllocation/ResourceAllocationController.cs [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+
+            var command = new Logic.Commands.ResourceAllocationRequest.Create(projectIdentifier.ProjectId)
+                .WithDiscipline(request.Discipline)
+                .WithType($"{request.Type}")
+                .WithProposedPerson(request.ProposedPersonId)
+                .WithOrgPosition(request.OrgPositionId)
+                .WithProposedChanges(request.ProposedChanges)
+                .WithIsDraft(request.IsDraft)
+                .WithAdditionalNode(request.AdditionalNote)
+                .WithPosition(request.OrgPositionInstance.Id, request.OrgPositionInstance.AppliesFrom,
```
[the remark this answers is not in the document]
Comment (xssic) [reply src/backend/api/Fusion.Resources.Api/Controllers/ResourceAllocation/ResourceAllocationController.cs]: WithPositionInstance

**Review by leifbjarte — APPROVED** [review]

**Pull request merged**